public/app/features/alerting/unified/RuleList.tsx

```diff
@@ -61,11 +61,19 @@ const RuleList = withErrorBoundary(
     );
 
     const promRequests = Object.entries(promRuleRequests);
+    const rulerRequests = Object.entries(rulerRuleRequests);
+
     const allPromLoaded = promRequests.every(
       ([_, state]) => state.dispatched && (state?.result !== undefined || state?.error !== undefined)
     );
     const allPromEmpty = promRequests.every(([_, state]) => state.dispatched && state?.result?.length === 0);
 
+    const allRulerEmpty = rulerRequests.every(([_, state]) => {
+      const rulerRules = Object.entries(state?.result ?? {});
+      const noRules = rulerRules.every(([_, result]) => result?.length === 0);
```

Comment:
Should we check `state.loading`, `state.dispatched` or `state.error`? This condition doesn't take request states into account, so if all requests are in progress, it will return true. Am I correct?

Reply:
you're right!

Reply:
done!